doc/arch/adr-006-transition-whitehall-to-publishing-api-index.md

```diff
@@ -0,0 +1,86 @@
+from publishing API.
+
+See [ADR 006](adr-004-transition-mainstream-to-publishing-api-index.md) for more
+details about moving mainstream content.
+
+This leaves two indexes that are populated the old way: `government` and `detailed`.
+
+`detailed` contains detailed guides and `government` contains everything else
+published by Whitehall publisher.
+
+We also have a separate worker (`publishing-queue-listener`) that listens to
+`*.links` notifications from publishing API, and *updates* the old indexes
+(but doesn't add any new content).
+
+## Decision
+
+We intend to get rid of the `government` and `detailed` indexes. Rummager
+should index Whitehall Publisher documents into the `govuk` index.
+
+All search indexing code should be removed from Whitehall Publisher.
+
+The old indexing code should be removed from Rummager:
+
+- Anything in `lib/indexer`
+- The `publishing-queue-listener` worker
+
+The [infrastructure to log indexing requests to disk](https://docs.publishing.service.gov.uk/manual/rummager-traffic-replay.html) can also be removed,
+since data from the publishing API can be resent at any time.
+
+## Consequences
+
+There are over 200,000 documents published by Whitehall Publisher.
+
+Making this change will enable us to treat all documents the same way in search,
+and we can make better use of the data we have about content to improve search,
+because there will be a single indexing process that we can change easily.
+
+For example, having this data in search would allows us to return more relevant results and/or improve the user experience of using search:
+
+- The **state** of a document, based on archiving policy and content audits
+- Different ways of grouping **document types** around user need
+- More **publishing history** details, besides the `public_updated_at` value
+- Information about **who** the content is applicable to or **where** it is relevant
+- Extensions to the topic taxonomy, like what **services** or **task lists** the content belongs to
+- Entities extracted from the text of the page (or its attachments), like **people**, **organisations**, **places**, **deadlines** and **forms**.
+
+Combining all of the indexes into a single index will change the document frequency statistics, which affects how search results are scored.
+
+## Action plan
+
+### Copy over documents as-is and test
+Before making any changes to Whitehall Publisher, we should copy all documents from the `government` and `detailed` indexes to the `govuk` index.
+
+These documents will be filtered out of search results, but will have an indirect impact on scoring of other documents.
+
+We can then run the search healthcheck to understand the impact of this. At this stage we may want to reevaluate how much each format gets boosted in the search query.
+
+Once this is done, the `govuk` indexing process can be extended to Whitehall Publisher formats. Steps 1-3 should be repeated for each document type or group of related document types.
```

Comment:
Which steps 1-3? I thought you might mean the [reindexing steps](https://github.com/alphagov/rummager/blob/master/doc/new-indexing-process.md#get-the-data-in-sync-on-integration) but we'd have to run all of them, not just 1-3.

Reply:
:man_facepalming: this refers to something I wrote before I remembered that doc exists